packages/edit-navigation/src/components/secondary-sidebar/inserter-sidebar.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { close } from '@wordpress/icons';
+import {
+	__experimentalLibrary as Library,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useViewportMatch } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editNavigationStore } from '../../store';
+import { useNavigationEditorRootBlock } from '../../hooks';
+
+function InserterSidebar() {
+	const SHOW_PREVIEWS = false;
+
+	const isMobileViewport = useViewportMatch( 'medium', '<' );
+
+	const { clientId: navBlockClientId } = useNavigationEditorRootBlock();
+
+	const { hasInserterItems, selectedBlockClientId } = useSelect(
+		( select ) => {
+			return {
+				hasInserterItems: select( blockEditorStore ).hasInserterItems(
+					navBlockClientId
+				),
+				selectedBlockClientId: select(
+					blockEditorStore
+				).getSelectedBlock()?.clientId,
+			};
+		},
+		[]
+	);
+
+	const { setIsInserterOpened } = useDispatch( editNavigationStore );
+
+	// Only concerned with whether there are items to display. If not then
+	// we shouldn't render.
+	if ( ! hasInserterItems ) {
+		return null;
+	}
+
+	return (
+		<div className="edit-navigation-layout__inserter-panel">
+			<div className="edit-navigation-layout__inserter-panel-header">
+				<Button
+					icon={ close }
+					onClick={ () => setIsInserterOpened( false ) }
+				/>
+			</div>
+			<div className="edit-navigation-layout__inserter-panel-content">
+				<Library
+					shouldFocusBlock={ isMobileViewport }
+					rootClientId={ navBlockClientId }
+					clientId={
+						navBlockClientId === selectedBlockClientId
+							? navBlockClientId
+							: undefined
+					}
+					showInserterHelpPanel={ SHOW_PREVIEWS }
+					showSearch={ false }
```

Comment:
Why is search hidden here? Isn't the plan to have more core blocks, like social icons, search etc and third party blocks in this inserter? Meaning that search might be useful. Could be only hide the search, if there are more than 8 blocks or something?

Reply:
Yeah, I agree with @spacedmonkey's comment.

Also have to consider the variations that get created from taxonomies and custom post types which could inflate the available blocks quite a bit.

Reply:
Good question to ask!

I've hidden it mainly because as a default experience it seemed completely OTT to have a search. 

Thinking again:

* Your point about more blocks in the future is valid. I considered it but decided we could conditionally reintroduce it later if necessary. I might be overcomplicating things I think.
* Users of assistive tech _might_ still rely on search to find blocks.

Either way [I've extracted the changes requires to disable search in the block library to a separate PR](https://github.com/WordPress/gutenberg/pull/34656) and will remove those commits from this PR so we don't have to consider it now.

I still think the search is visually very dominant given that by default we have a limited number of blocks. But I don't feel super strongly about it.

Reply:
Update: I removed the changes from this PR that hide the search bar. If we need to deal with that UX then we can handle in a follow up.